### ISSUE TITLE
list indices must be integers or slices, not str

### ISSUE THREAD

**Issue by huutuan1705** [explainingai-code/FasterRCNN-PyTorch#3]
Hello, 

I see your code in the internet, and this is so interesting. I cloned it and used for my project. I try changing batch_size from 1 to 4 and backbone from vgg16 to resnet101. But I have a problem. You can see this image to know about my error:

![image](https://github.com/user-attachments/assets/110a4cf4-0f6d-4e8f-92bf-45b9810ce210)

I think the cause of this issue is that I changed the batch_size, which led to changes in the list of boxes. I try to fix this problem for three days, but not success. I really need your help. Can you help me fix that. Here is my code:

https://colab.research.google.com/drive/1YpICgRMK7wiIFn85dt753j18NIY3JgfH?usp=sharing

Thank you so much!!!

**Comment by explainingai-code** [comment]
Hello @huutuan1705 ,

This repo is only made for single batch size. If you want to train with larger than 1 batch size, then you can use gradient accumulation for that purpose.  So batch size would still be 1 but network would be updated with accumulated gradients over multiple batches.
Assuming you want to train with 4 batch size, so just set the `acc_steps` parameter in config to be 4
https://github.com/explainingai-code/FasterRCNN-PyTorch/blob/main/config/voc.yaml#L37

**Comment by huutuan1705** [comment]
Thank you so much, I understand that. Thanks for your help.
